components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIManagerFactory.java

```diff
@@ -180,6 +195,38 @@ public APIPublisher getAPIProvider(String username) throws APIManagementExceptio
     }
 
     /**
+     * Get Analyzer for a particular user
+     *
+     * @param username The username of user who's requesting the object
+     * @return APIPublisher object
+     * @throws APIManagementException if error occurred while initializing Analytics object
+     */
+    public Analyzer getAnalyzer(String username) throws APIManagementException {
+        Analyzer analyzer = analyzers.get(username);
+        if (analyzer == null) {
+            synchronized (username.intern()) {
+                analyzer = analyzers.get(username);
+                if (analyzer != null) {
+                    return analyzer;
+                }
+
+                analyzer = newAnalyzer(username);
```

Comment:
for consumer and provider we have "newConsumer" and "newProvider". I was sticking to the convention. WDYT?